src/MDMScheduler.m

```diff
@@ -81,16 +81,33 @@ - (MDMSchedulerActivityState)activityState {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan to:(id)target {
+  [self commonAddPlan:plan toTarget:target log:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan named:(NSString *)name to:(id)target {
+  NSParameterAssert(name.length > 0);
+  [self commonAddPlan:plan toTarget:target log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:target name:name removal:FALSE]];
```

Comment:
Interesting. I think the reason I would hesitate to do that is because Plan is a client facing public object that the client may keep around. And these metadata are private and internal and I'd rather we keep track of it internally.